framework/src/auxkernels/MaterialStdVectorAux.C

```diff
@@ -34,5 +37,8 @@ MaterialStdVectorAux::~MaterialStdVectorAux()
 Real
 MaterialStdVectorAux::getRealValue()
 {
+  mooseAssert(_special_qp < _q_point.size(), "MaterialStdVectorAux.  special_qp specified as " << _special_qp << " but there are only " << _q_point.size() << " quadpoints in the element");
```

Comment:
I wouldn't use NaN.  Both for reasons like the above, and because some architectures might not implement quiet_NaN() and therefore would raise a floating point exception, when all you are trying to do is indicate an invalid/error state without crashing (if I understand correctly).